habitat/tasks/rearrange/multi_task/composite_task.py

```diff
@@ -167,7 +167,8 @@ def reset(self, episode: Episode):
         self._inf_cur_node = 0
         self._inf_cur_task = None
         self._get_next_inf_sol(episode)
-        return super().reset(episode)
+        self.cached_tasks = {}
```

Comment:
Clearing the cache here or the target_idx will not be reset properly on line 199 when the environment resets